src/modules/App.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import Board from './Board';
+import Header from './Header';
+import Footer from './Footer';
+import PopUp from './PopUp';
+import { TransitionAlertsWin, TransitionAlertsLose } from './Alert';
+import { getInitialBoardArray, getValueFromLocalStorage } from './utils/helpers';
+
+const App = () => {
+  const [boardSize, setBoardSize] = useState(4);
+  const [bestScore, setBestScore] = useState(getValueFromLocalStorage('bestScore') || [0]);
+  const localHistory = getValueFromLocalStorage('2048-history');
+  const isLocalBoardSizeSame =
+    localHistory && localHistory[localHistory.length - 1].board.length === boardSize * boardSize;
+  const [history, setHistory] = useState(
+    localHistory && isLocalBoardSizeSame
+      ? localHistory
+      : [
+          {
+            board: getInitialBoardArray(boardSize * boardSize),
+            score: 0,
+            win: false,
+            gameOver: false,
+          },
+        ]
+  );
+  const [isPopUpActive, setIsPopUpActive] = useState(false);
+  const [openWin, setOpenWin] = useState(false);
+  const [openLose, setOpenLose] = useState(false);
+  const [winningNumber, setWinningNumber] = useState('2048');
```

Comment:
magic number?

Reply:
well, I thought it's not that magic since winningNumber, setWinningNumber right in front of it, am I wrong?

Reply:
What if you want to change the system for  scoring this game (let’s say 3^n instead of 2^n), and you need to go through the whole project and manually change it... If you set this number in a constant and used it throughout the app, then you’ll need to change only it in 1 place... I would suggest using constants instead of strings/numbers because it’s less probability to make a typo

Reply:
ok, I won't make same mistakes again